### PR TITLE
[SYSTEMML-1744] JCuda jars in extra assembly jar

### PR DESCRIPTION
**Pull request by nakul02**
Ping @niketanpansare - can you please test, if possible, whether this works for a pip install?
Ping @deroneriksson, @gweidner  - can you please review this?

### CODE REVIEW

**Comment by nakul02** [comment]
Something to be aware of - This includes the platform specific cuda jars for which SystemML was built on. 

In an offline discussion with @niketanpansare, he said it was ok to have Linux jcuda native jars, because that would probably be the more widely used platform. For other platforms, we could provide instructions for their setup.

Another option would be to just include ALL or.jcuda jars for all platforms. This would blow up the `extra` jar size. Also, I'd need to test if that would work.

**Comment by niketanpansare** [comment]
@nakul02 The jcuda jars are included likely as resource rather than `compile` by this PR.  Also, I don't see jcuda classes. Here is the content of `extra` jar:
```
lib/
lib/jcuda-0.8.0.jar
org/apache/sysml/api/dl/BaseDMLGenerator$$anonfun$invoke$2.class
org/apache/sysml/api/dl/Caffe2DML$$anonfun$org$apache$sysml$api$dl$Caffe2DML$$backward$1.class
org/apache/sysml/api/dl/Caffe2DMLModel$$anonfun$getPredictionScript$8.class
org/apache/sysml/api/dl/CaffeNetwork$.class
org/apache/sysml/api/dl/DMLGenerator$$anonfun$initWeights$1.class
org/apache/sysml/api/dl/HasWeight$class.class
org/apache/sysml/api/dl/IsLossLayer.class
org/apache/sysml/api/dl/LearningRatePolicy.class
org/apache/sysml/api/dl/Nesterov$.class
org/apache/sysml/api/dl/Nesterov.class
...
```

Hence,  the following python code throws an error `java.lang.ClassNotFoundException: jcuda.runtime.JCuda`:
```python
import numpy as np
from sklearn import datasets
from systemml.mllearn import LinearRegression
# Load the diabetes dataset
diabetes = datasets.load_diabetes()
# Use only one feature
diabetes_X = diabetes.data[:, np.newaxis, 2]
# Split the data into training/testing sets
X_train = diabetes_X[:-20]
X_test = diabetes_X[-20:]
# Split the targets into training/testing sets
y_train = diabetes.target[:-20]
y_test = diabetes.target[-20:]
# Create linear regression object
regr = LinearRegression(spark, fit_intercept=True, C=float("inf"), solver='direct-solve').setGPU(True).setForceGPU(True).setStatistics(True)
# Train the model using the training sets
regr.fit(X_train, y_train)
```

File: src/assembly/extra/LICENSE [hunk 1]
```diff
@@ -460,3 +460,51 @@ Copyright 2017 The TensorFlow Authors.  All rights reserved.
...
    See the License for the specific language governing permissions and
    limitations under the License.
+
+===============================================================================
+
+The following compile-scope dependencies come under the MIT License
+
+JCuda (jcuda.org)
+
+org.jcuda.jcuda:0.8.0
```
Comment (deroneriksson) [inline src/assembly/extra/LICENSE]: Please update these to reflect maven groupId:artifactId:version, so `org.jcuda.jcuda:0.8.0` is `org.jcuda:jcuda:0.8.0`, etc.

**Comment by deroneriksson** [comment]
I see the following after `mvn clean package -P distribution`: 
image

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1700/

**Comment by nakul02** [comment]
@niketanpansare - can you please check now?

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1705/

**Comment by niketanpansare** [comment]
LGTM. Thanks @nakul02 :+1:  ... I would recommend to wait until you get LGTM from @deroneriksson or @gweidner

**Comment by nakul02** [comment]
Thanks @niketanpansare 
Will do.

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1714/

**Comment by deroneriksson** [comment]
LGTM. Thank you @nakul02.

For reference, on GitHub, the 0.8.0 tag does not include the MIT LICENSE file that exists on the master branch. However, based on the discussion at https://github.com/jcuda/jcuda-main/issues/18, 0.8.0 is indeed MIT licensed and the license file is at http://jcuda.org/License.txt.

**Comment by nakul02** [comment]
Thanks @deroneriksson, I shall merge.

**Pull request closed without merging**